cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/StudioService.java

```diff
@@ -40,5 +61,51 @@ protected void configure() {
     createTable(DraftStore.TABLE_SPEC);
     createTable(OAuthStore.TABLE_SPEC);
     createTable(ConnectionStore.CONNECTION_TABLE_SPEC);
+    setProperties(Collections.singletonMap(CONNECTION_TYPE_CONFIG, GSON.toJson(connectionConfig)));
+  }
+
+  @Override
+  public void initialize(SystemServiceContext context) throws Exception {
+    String connConfig = context.getSpecification().getProperty(CONNECTION_TYPE_CONFIG);
+    connectionConfig = GSON.fromJson(connConfig, ConnectionConfig.class);
+    // only do the connection creation on first instance to avoid transaction conflict
+    if (context.getInstanceId() != 0 || connectionConfig == null || connectionConfig.getConnections().isEmpty()) {
+      return;
+    }
+
+    try {
+      createPreconfiguredConnections(context);
+    } catch (Exception e) {
+      // we don't want the service fail to start
+      LOG.error("Error creating the preconfigured connections");
+    }
+  }
+
+  private void createPreconfiguredConnections(SystemServiceContext context) throws IOException {
+    ConnectionStore connectionStore = new ConnectionStore(context);
+    for (PreconfiguredConnectionCreationRequest creationRequest : connectionConfig.getConnections()) {
+      if (creationRequest.getName() == null || creationRequest.getNamespace() == null) {
+        continue;
+      }
+
+      NamespaceSummary namespaceSummary = context.getAdmin().getNamespaceSummary(creationRequest.getNamespace());
+      if (namespaceSummary == null) {
+        LOG.warn("Namespace {} does not exist, skipping creating connection {}", creationRequest.getNamespace(),
+                 creationRequest.getName());
+      }
+
+      ConnectionId connectionId = new ConnectionId(namespaceSummary, creationRequest.getName());
+      long now = System.currentTimeMillis();
+      Connection connectionInfo = new Connection(
+        creationRequest.getName(), connectionId.getConnectionId(), creationRequest.getPlugin().getName(),
+        creationRequest.getDescription(), true,
+        creationRequest.getName().equals(connectionConfig.getDefaultConnection()) ? true : false,
+        now, now, creationRequest.getPlugin());
+      try {
+        connectionStore.saveConnection(connectionId, connectionInfo, false);
```

Comment:
this is different behavior than before, where these connections will now be written to the store and persist, even if the default connections change. Is that intended?

Reply:
I remembered the existing behavior incorrectly, ignore this comment.